cpp_api/api.h

```diff
@@ -0,0 +1,24 @@
+#include <vector>
+
+#include <goto-programs/goto_model.h>
+
+#include <util/message.h>
+#include <util/options.h>
+#include <util/config.h>
```

Comment:
We need to be very careful which data structures should be exposed. It doesn't seem to be a good idea to expose things like `goto_modelt`, `message_handlert`, etc. We should expose the operations that the client needs, but not give access to the objects themselves if it can be avoided. I guess this needs to be closely reviewed as we make progress and learn more about the integration.